frontend/__tests__/utils/seo-utils.test.ts

```diff
@@ -0,0 +1,58 @@
+import { renderHook } from '@testing-library/react';
+
+import { useLocation } from '@remix-run/react';
+
+import { i18n } from 'i18next';
+import { useTranslation } from 'react-i18next';
+import { afterEach, describe, expect, it, vi } from 'vitest';
+
+import { getClientEnv } from '~/utils/env-utils';
+import { useI18nNamespaces } from '~/utils/route-utils';
+import { useCanonicalURL } from '~/utils/seo-utils';
+
+vi.mock('@remix-run/react');
+vi.mock('react-i18next');
+vi.mock('~/utils/env-utils');
+vi.mock('~/utils/route-utils');
+
+describe('useCanonicalURL', () => {
+  afterEach(() => {
+    vi.clearAllMocks();
```

Comment:
This should be `vi.resetAllMocks()`, since you're mocking your returns inside of each test.

Reply:
Fixed